DataPlotly/core/plot_factory.py

```diff
@@ -270,6 +272,17 @@ def add_source_field_or_expression(field_or_expression):
                                                                               context, default_value)
                 stroke_colors.append(value.name())
 
+        # build up color scale for legend (if possible!)
+        if isinstance(self.settings.data_defined_properties.property(PlotSettings.PROPERTY_COLOR).transformer(),
```

Comment:
I'm pretty sure the reason is that the legend wants as mandatory the first value with index 0 and the last with index 1. If the last value is not set as 1 index then nothing is rendered.

Reply:
I've checked the contents of the array though, and it looks fine to me ...

Reply:
Actually the current array contains a list of hex color in the `colors` parameter while it should contain a list of numbers, then the color_scale will work. We need to grab the field (or expression) from the assistant dialog, evaluate it and get a list of *real* values.

Reply:
Oh, so you can't mix and match the two? Hmmmm, this gets incredibly complex now....

Reply:
Seems not. Having real numbers for the color and the [index, hex] for the colors scale seems the best way to achieve it. 

Is it really getting so complex?

Reply:
Yes -- because the color mapping used in the color ramp assistant isn't necessarily linear!

Reply:
Maybe we just drop the legend option?

Reply:
Well given that we are having a lot of new features thanks to @SGroe and some other fixes, I'll say that we can drop the legend for the moment and writing an issue as reminder. What is your opinion?